### PR TITLE
Fix `resize_mask` to honor `NEAREST_EXACT` interpolation

### LINKED ISSUE #9188: InterpolationMode.NEAREST_EXACT and InterpolationMode.NEAREST are the same for tv_tensors.Mask
### 🐛 Describe the bug

When resizing tensor wrapped in tv_tensors.Mask interpolation mode NEAREST_EXACT is changed to NEAREST

```python
import torch
import torch.nn.functional as F
import torchvision.transforms.v2 as T
import torchvision.transforms.v2.functional as TF
from torchvision import tv_tensors


def compare(mask, size):
    mask_nearest = TF.resize(mask, size=size, interpolation=T.InterpolationMode.NEAREST)
    mask_nearest_exact = TF.resize(mask, size=size, interpolation=T.InterpolationMode.NEAREST_EXACT)
    mask_nearest_interpolate = F.interpolate(mask.unsqueeze(0), size=size, mode='nearest').squeeze(0)
    mask_nearest_exact_interpolate = F.interpolate(mask.unsqueeze(0), size=size, mode='nearest-exact').squeeze(0)
    
    print(torch.equal(mask_nearest, mask_nearest_exact))
    print(torch.equal(mask_nearest_interpolate, mask_nearest_exact_interpolate))
    print(torch.equal(mask_nearest, mask_nearest_interpolate))
    print(torch.equal(mask_nearest_exact, mask_nearest_exact_interpolate))
    

height = 64
width = 64
new_height = 16
new_width = 16
mask = torch.randint(low=0, high=2, size=(1, height, width)).float()

compare(mask, (new_height, new_width))
print()

mask = tv_tensors.Mask(mask)
compare(mask, (new_height, new_width))
```

```
False
False
True
True

True
False
True
False
```

### Versions

torch = "2.8.0"
torchvision = "0.23.0"

### PR DESCRIPTION
**Pull request by joannemiki57**
Closes #9188.

The `interpolation` argument was effectively ignored for `tv_tensors.Mask` inputs. `resize_mask` hardcoded `InterpolationMode.NEAREST` when calling `resize_image`, so `F.resize(mask, size, interpolation=InterpolationMode.NEAREST_EXACT)` produced the same output as plain `NEAREST`, contrary to what the docs imply.

**Reproduction (before this PR):**

```python
import torch
from torchvision import tv_tensors
from torchvision.transforms.v2 import functional as F, InterpolationMode

mask = tv_tensors.Mask(torch.randint(0, 4, (1, 7, 11)))
a = F.resize(mask, size=[14, 22], interpolation=InterpolationMode.NEAREST)
b = F.resize(mask, size=[14, 22], interpolation=InterpolationMode.NEAREST_EXACT)
assert torch.equal(a, b)   # passes — bug: NEAREST_EXACT was a no-op
```

After this PR, `b` is computed with `nearest-exact` and the two tensors differ on inputs where the two algorithms disagree.

### What changed

- Add an `interpolation` parameter to `resize_mask` and `_resize_mask_dispatch`, in the position expected by `F.resize` (`mask, size, interpolation, max_size`) and with the same annotation and default (`Union[str, InterpolationMode, int] = InterpolationMode.BILINEAR`) so `test_functional_signature[F.resize_mask-Mask]` passes.
- Validate through the existing `_check_interpolation` helper. Anything other than `NEAREST` / `NEAREST_EXACT` is silently coerced to `NEAREST`, preserving the historical behavior of callers that previously relied on the argument being ignored. Comparisons are chained `!=` / `and` (no set literals) so the function stays TorchScript-compatible.
- Extend `TestResize.test_kernel_mask` to parametrize over `NEAREST` and `NEAREST_EXACT`, exercising eager / scripted / batched / CUDA via `check_kernel`.

A prior attempt at the same fix is open as #9384; the maintainer review there spelled out the signature ordering, set-literal, and default-value issues that this PR addresses from scratch.

### Test plan

- `pytest test/test_transforms_v2.py -k "TestResize and (test_kernel_mask or test_functional_signature)" -v` covers the new parametrization and the signature contract (to be exercised on CI).
- Existing call sites of `resize_mask` (`resized_crop_mask` at `_geometry.py:2845`) pass `(mask, size)` positionally and are backward-compatible with the new signature.

cc @vfdev-5

### CODE REVIEW

**Review by NicolasHug — APPROVED** [review]
Thanks for the fix @joannemiki57
